src/oracles/CoinbaseOracle.sol

```diff
@@ -0,0 +1,120 @@
+
+    /**
+     * @notice Constructor to initialize the CoinbaseOracle contract.
+     * @param _symbol The symbol of the asset being reported.
+     * @param _decimals The number of decimals in the reported price.
+     * @param _reporter The address of the reporter allowed to push price data.
+     */
+    constructor(string memory _symbol, uint8 _decimals, address _reporter) {
+        symbol = _symbol;
+        decimals = _decimals;
+        reporter = _reporter;
+    }
+
+    /**
+     * @notice Returns the latest round data.
+     * @return roundId The ID of the latest round.
+     * @return answer The latest price.
+     * @return startedAt The timestamp when the round started.
+     * @return updatedAt The timestamp when the round was updated.
+     * @return answeredInRound The round ID in which the answer was computed.
+     */
+    function latestRoundData()
+        external
+        view
+        override
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        int256 latestAnswer = roundAnswers[lastRoundId];
+        uint256 latestTimestamp = roundTimestamps[lastRoundId];
+        return (lastRoundId, latestAnswer, latestTimestamp, latestTimestamp, lastRoundId);
+    }
+
+    /**
+     * @notice Returns the data for a specific round.
+     * @param _roundId The round ID to retrieve the data for.
+     * @return roundId The ID of the round.
+     * @return answer The price of the round.
+     * @return startedAt The timestamp when the round started.
+     * @return updatedAt The timestamp when the round was updated.
+     * @return answeredInRound The round ID in which the answer was computed.
+     */
+    function getRoundData(uint80 _roundId)
+        external
+        view
+        override
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        int256 latestAnswer = roundAnswers[_roundId];
+        uint256 latestTimestamp = roundTimestamps[_roundId];
+        return (_roundId, latestAnswer, latestTimestamp, latestTimestamp, _roundId);
+    }
+
+    /**
+     * @notice Pushes a new price to the oracle.
+     * @param priceData The encoded price data.
+     * @param signature The signature to verify the authenticity of the data.
+     */
+    function pushPrice(bytes memory priceData, bytes memory signature) external {
+        (
+            string memory kind, // e.g. "price"
+            uint256 timestamp, // e.g. 1629350000
+            string memory ticker, // e.g. "BTC"
+            uint256 price // 6 decimals
+        ) = abi.decode(priceData, (string, uint256, string, uint256));
+
+        uint256 latestTimestamp = roundTimestamps[lastRoundId];
+
+        require(keccak256(abi.encodePacked(kind)) == keccak256(abi.encodePacked("price")), "Invalid kind.");
+        require(price < uint256(type(int256).max), "Price exceeds max value.");
```

Comment:
Given we store these as `int256` why do we need this check?

Reply:
I think we need this as `uint256(type(int256).max < type(uint256).max`

Reply:
Removing the check will now cause overflow but if you push `uint256(type(int256).max) + 1`

Then (, int256 answer, , , ) = coinbaseOracle.latestRoundData(ethTicker);

answer is : `-57896044618658097711785492504343953926634992332820282019728792003956564819968`

But it's a weird scenario so I'm happy to remove the check. 

Do you agree @Reinis-FRP @mrice32 @chrismaree?

Reply:
i still tvink its safer to keep the check

Reply:
I agree that keeping the check is safer. Whenever we cast externally sourced data to a type that can produce overflow, we should check for the overflow.